### PR TITLE
Clarify U2F attestation verification instructions

### LINKED ISSUE #857: Clarify U2F attestation verification
Step 4 of the [U2F attestation verification procedure](https://w3c.github.io/webauthn/#fido-u2f-attestation) could use some clarification:

> Convert the COSE_KEY formatted credentialPublicKey (see Section 7 of [RFC8152]) to CTAP1/U2F public Key format [FIDO-CTAP].

Suggest adding "Section XYZ of [FIDO-CTAP]. CTAP is a big spec and it's not immediately clear that there's actually any section that defines a public key format.

>    * Let publicKeyU2F represent the result of the conversion operation and set its first byte to 0x04. Note: This signifies uncompressed ECC key format.
>    * Extract the value corresponding to the "-2" key (representing x coordinate) from credentialPublicKey, confirm its size to be of 32 bytes and concatenate it with publicKeyU2F. If size differs or "-2" key is not found, terminate this algorithm and return an appropriate error.
>    * Extract the value corresponding to the "-3" key (representing y coordinate) from credentialPublicKey, confirm its size to be of 32 bytes and concatenate it with publicKeyU2F. If size differs or "-3" key is not found, terminate this algorithm and return an appropriate error.

I think a final bullet should be added stating `Let publicKeyU2F be the concatenation of (0x04 || x || y).`

### PR DESCRIPTION
**Pull request by emlun**
This fixes #857.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/861.html" title="Last updated on May 2, 2018, 11:00 AM GMT (3afbcf9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/861/aabeeaf...3afbcf9.html" title="Last updated on May 2, 2018, 11:00 AM GMT (3afbcf9)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -3537,19 +3537,17 @@ This attestation statement format is used with FIDO U2F authenticators using the
...
-    1. Convert the COSE_KEY formatted |credentialPublicKey| (see Section 7 of [[!RFC8152]]) to CTAP1/U2F public Key
-        format [[!FIDO-CTAP]]. 
-        - Let |publicKeyU2F| represent the result of the 
-            conversion operation and set its first byte to 0x04.
-            Note: This signifies uncompressed ECC key format.
-        - Extract the value corresponding to the "-2" key 
-            (representing x coordinate) from |credentialPublicKey|, 
-            confirm its size to be of 32 bytes and concatenate it with |publicKeyU2F|.
+    1. Convert the COSE_KEY formatted |credentialPublicKey| (see Section 7 of [[!RFC8152]]) to CTAP1/U2F user public key
+        format (see Section 4.3 of [[!FIDO-U2F-Message-Formats]]).
```
Comment (equalsJeffH) [inline index.bs]: We can link directly to both Section 7 of [[!RFC8152]] and Section 4.3 of [[!FIDO-U2F-Message-Formats]], tho see issue #872 (i would make entries in `<pre class="anchors">` to do so).

File: index.bs [hunk 2]
```diff
@@ -3537,19 +3537,17 @@ This attestation statement format is used with FIDO U2F authenticators using the
...
-            (representing y coordinate) from |credentialPublicKey|, 
-            confirm its size to be of 32 bytes and concatenate it with |publicKeyU2F|.
+        - Let |y| be the value corresponding to the "-3" key (representing y coordinate) in |credentialPublicKey|, and confirm its
+            size to be of 32 bytes.
             If size differs or "-3" key is not found, terminate this algorithm and return an appropriate error. 
+        - Let |publicKeyU2F| be the concatenation <code>0x04 || |x| || |y|</code>.
+
+            Note: This signifies uncompressed ECC key format.
     1. Let |verificationData| be the concatenation of (0x00 || |rpIdHash| ||
         |clientDataHash| || |credentialId| || |publicKeyU2F|) (see Section 4.3 of [[!FIDO-U2F-Message-Formats]]).
```
Comment (equalsJeffH) [inline index.bs]: ditto

File: index.bs (hunk 1 above)
Comment (emlun) [reply index.bs]: Ok. I made an attempt at this in PR #883, does that look okay?

File: index.bs (hunk 2 above)
Comment (emlun) [reply index.bs]: ditto

File: index.bs (hunk 1 above)
Comment (herrjemand) [reply index.bs]: @emlun maybe "1. Convert the COSE_KEY formatted |credentialPublicKey| (see Section 7 of [[!RFC8152]]) to Raw ANSI X9.62 public key format(see ALG_KEY_ECC_X962_RAW  in Section 3.6.2 of [[!FIDO-Registry]]).

Comment (equalsJeffH) [reply index.bs]: @emlun 
> k. I made an attempt at this in PR #883, does that look okay?
yes, see https://github.com/w3c/webauthn/pull/883#pullrequestreview-116424391

Comment (equalsJeffH) [reply index.bs]: @herrjemand's suggestion is also fine by me and we can use the same technique as in #883 to link directly to Section 3.6.2 of [[!FIDO-Registry]]).  either works for me.

**Comment by equalsJeffH** [comment]
LGTM, just pick ref'g the U2F spec or @herrjemand's suggestion of ref'g FIDO registry and we're good. thx :)

File: index.bs (hunk 1 above)
Comment (emlun) [reply index.bs]: `[[!FIDO-Registry]]` doesn't seem to be defined in the spec yet - what does it refer to?

**Comment by equalsJeffH** [comment]
merge from master ought to now fix breaking build

File: index.bs (hunk 1 above)
Comment (herrjemand) [reply index.bs]: @emlun To this https://drafts.fidoalliance.org/fido-2/latest/fido-registry-v2.0-wd-20180426.html *)

**Comment by nadalin** [comment]
@emlun Please fix and merge

**Comment by emlun** [comment]
Continuing from https://github.com/w3c/webauthn/pull/861#discussion_r185292892:

@herrjemand Thanks! However it might be prudent to not reference that document at this time since, it's a draft that explicitly states it "[...]  is not intended to be a basis for any implementations as the Specification may change". Please correct me if I'm wrong here. For now I'll merge this PR in its current state, and I'll open a new issue to perhaps update this reference later.

**Comment by emlun** [comment]
To be continued in #891.

**Pull request merged**